src/index.ts

```diff
@@ -163,6 +163,25 @@ function FlatpickrInstance(
     }
   }
 
+  function getHourElementValueAsMilitary24(): number {
+    const hourElementNumberValue = self.hourElement
+      ? +self.hourElement.value
+      : self.config.defaultHour;
+
+    return self.amPM !== undefined && hourElementNumberValue
+      ? ampm2military(hourElementNumberValue, self.amPM.textContent as string)
+      : hourElementNumberValue;
```

Comment:
Won't `hourElementNumberValue ` always be true ?

Reply:
It could be equal to 0 thus false and we would return 0